core/src/main/scala/scalaz/IList.scala

```diff
@@ -2,18 +2,19 @@ package scalaz
 
 import scala.annotation.tailrec
 import scala.annotation.unchecked.uncheckedVariance
-import std.option.{ cata, none, some }
-import std.stream.{ toZipper => sToZipper }
-import std.tuple.{ tuple2Bitraverse => BFT }
-import Liskov.{ <~<, refl }
-import IList.{empty, single}
+import std.option.{cata, none, some}
+import std.stream.{toZipper => sToZipper}
+import std.tuple.{tuple2Bitraverse => BFT}
+import Liskov.{<~<, refl}
+import IList.{empty, foldRightLazy, single, byNeed}
+import scalaz.std.option
 
 /**
  * Safe, invariant alternative to stdlib `List`. Most methods on `List` have a sensible equivalent
  * here, either on the `IList` interface itself or via typeclass instances (which are the same as
  * those defined for stdlib `List`). All methods are total and stack-safe.
  */
-sealed abstract class IList[A] extends Product with Serializable {
+trait IList[A] extends Product with Serializable {
```

Comment:
@xuwei-k is the binary-compatibility advantage of abstract class real if client code does not implement IList? because otherwise making the trait `sealed` should be enough... maybe?